### PR TITLE
Add support for data object

### PR DESCRIPTION
**Pull request by hrach**

### CODE REVIEW

File: src/keywords.YAML-tmLanguage [hunk 1]
```diff
@@ -44,7 +44,7 @@ repository:
                 match: \b(constructor|init)\b
                 name: entity.name.function.constructor
             -
-                match: \b(companion|object)\b
+                match: \b(companion|object|data\s+object)\b
```
Comment (nishtahir) [inline src/keywords.YAML-tmLanguage:47]: Thanks for your contribution. I'm assuming that this change is motivated by https://kotlinlang.org/docs/object-declarations.html#data-objects. I'd rather have `data` tagged independently as a storage modifier than make `data object` as a single keyword. I think the correct place to make the adjustment would be making the change [here](https://github.com/nishtahir/language-kotlin/blob/0553adec020f350bb726063a4bf85f587aac673d/src/classes.YAML-tmLanguage#L5)

Comment (hrach) [reply src/keywords.YAML-tmLanguage:47]: Ok, I'll rework :)

Comment (hrach) [reply src/keywords.YAML-tmLanguage]: So I did so, hopefully, it is correct now.

Comment (hrach) [reply src/keywords.YAML-tmLanguage]: @nishtahir please :)

**Review by nishtahir — APPROVED** [review]
Thanks for your contribution 🎉. This looks good. It still needs a regression test and update to the corpus but I can add that

**Pull request merged**